diff -urN qtcurve-1.8.17/qt4/kwin/qtcurveclient.cpp qtcurve-1.8.17-patched/qt4/kwin/qtcurveclient.cpp
--- qtcurve-1.8.17/qt4/kwin/qtcurveclient.cpp	2013-10-18 23:35:20.000000000 +1100
+++ qtcurve-1.8.17-patched/qt4/kwin/qtcurveclient.cpp	2013-10-30 12:38:19.000000000 +1100
@@ -1487,7 +1487,7 @@
     {
         if (maximizeMode()==MaximizeFull)
         {
-            if (!options()->moveResizeMaximizedWindows() && itsResizeGrip)
+            if (Handler()->borderlessMax() && itsResizeGrip)
                 itsResizeGrip->hide();
         }
         else if (itsResizeGrip)
diff -urN qtcurve-1.8.17/qt4/kwin/qtcurveclient.h qtcurve-1.8.17-patched/qt4/kwin/qtcurveclient.h
--- qtcurve-1.8.17/qt4/kwin/qtcurveclient.h	2013-10-18 23:35:20.000000000 +1100
+++ qtcurve-1.8.17-patched/qt4/kwin/qtcurveclient.h	2013-10-30 12:37:56.000000000 +1100
@@ -75,8 +75,7 @@
     bool eventFilter(QObject *o, QEvent *e);
     bool isMaximized() const
         {
-            return maximizeMode() == MaximizeFull &&
-                !options()->moveResizeMaximizedWindows();
+            return maximizeMode() == MaximizeFull && Handler()->borderlessMax();
         }
     void menuBarSize(int size);
     void statusBarState(bool state);
