diff -urN qtcurve-1.8.18/qt4/kwin/qtcurveclient.cpp qtcurve-1.8.18-patched/qt4/kwin/qtcurveclient.cpp
--- qtcurve-1.8.18/qt4/kwin/qtcurveclient.cpp	2014-01-26 07:13:54.000000000 +1100
+++ qtcurve-1.8.18-patched/qt4/kwin/qtcurveclient.cpp	2014-07-05 16:17:12.513703255 +1100
@@ -1486,7 +1486,7 @@
     {
         if (maximizeMode()==MaximizeFull)
         {
-            if (!options()->moveResizeMaximizedWindows() && m_resizeGrip)
+            if (Handler()->borderlessMax() && m_resizeGrip)
                 m_resizeGrip->hide();
         }
         else if (m_resizeGrip)
diff -urN qtcurve-1.8.18/qt4/kwin/qtcurveclient.h qtcurve-1.8.18-patched/qt4/kwin/qtcurveclient.h
--- qtcurve-1.8.18/qt4/kwin/qtcurveclient.h	2014-01-26 07:13:54.000000000 +1100
+++ qtcurve-1.8.18-patched/qt4/kwin/qtcurveclient.h	2014-07-05 16:18:34.853120883 +1100
@@ -79,8 +79,7 @@
     bool eventFilter(QObject *o, QEvent *e);
     bool isMaximized() const
     {
-        return (maximizeMode() == MaximizeFull &&
-                !options()->moveResizeMaximizedWindows());
+        return (maximizeMode() == MaximizeFull && Handler()->borderlessMax());
     }
     void menuBarSize(int size);
     void statusBarState(bool state);
